Simplify local develoment with docker compose

tancnle opened the pull request:
This PR helps to simplify local development with `docker-compose`. This is a follow-up from @macite's [comment](https://github.com/thoth-tech/splashkit-translator/pull/2#issuecomment-1090075108) on https://github.com/thoth-tech/splashkit-translator/pull/2.

## How to test

- Check out the branch
- Make sure Docker Desktop is installed
- Build a new Docker image 
  ```sh
  docker compose build
  ```
- Run the translator using the above image
   ```sh
  docker compose run --rm headerdoc python,csharp
  
  Translating SplashKit Core to python,csharp
  Executing python translator...
  Done!
  Output written!
  
  Executing csharp translator...
  Done!
  Output written!
  ```

tancnle committed:
feat: Simplify development with docker compose

This removes the needs of running a lengthy docker command that requires
volume map of the SplashKit code folder.

tancnle committed:
docs: update docker instructions

Update docker compose commands and minor formatting on Docker
instructions.

tancnle commented:
👋🏼 @dylanmedlin @allanfarrell could anyone of you test the docker-compose set up on a Windows machine 🙏🏼 ?

tancnle commented:
👋🏼 @macite could you help to review this PR, please 🙏🏼?

dylanmedlin commented:
I get a "No such file or directory" error when running on Windows. This could be due to symbolic links not working correctly with Windows

allanfarrell commented:
![image](https://user-images.githubusercontent.com/22616875/166125672-01b02496-297c-417e-9294-df29a9a0a732.png)

tancnle committed:
fix: call main script instead of symlink

This fixes issue calling the symlink `translate` on Windows.

tancnle commented:
Thanks for testing it @dylanmedlin and @allanfarrell. I have updated the PR (see [change](https://github.com/thoth-tech/splashkit-translator/pull/5/commits/c0850ea0bed940b423f211a60533976be85cc291)) to call the main script directly instead of symlink. Hopefully, this works for Windows 🤞🏼. Could you kindly do another test?

allanfarrell commented:
Still not working for me.
![image](https://user-images.githubusercontent.com/22616875/166127168-323b1131-47e9-4657-b8ce-d2fb7a60327d.png)

tancnle commented:
@allanfarrell  You might have already done this, have you pulled the latest change before re-running the test? 🤔

allanfarrell commented:
Yeah, I have done that.
![image](https://user-images.githubusercontent.com/22616875/166127331-cdf2e3c5-1890-407e-9eec-4f267f392ba7.png)

tancnle committed:
Merge branch 'master' into feature/docker-compose

* master:
  Create .gitattributes
  docs: update docker documentation
  fix: convert to linux line endings

tancnle commented:
> Yeah, I have done that.

@allanfarrell That's odd 🤔 I have merged the change from https://github.com/thoth-tech/splashkit-translator/pull/4 to address the line ending issue with Windows. Could you give it another go?

allanfarrell commented:
@tancnle I pulled the branch, rebuilt it and tried to run it. Got the same error.

tancnle committed:
fix: convert to shell script and use full path

Relative path to the wrapper script is not interpreted correctly in
Windows and displays 'No such file or directory error'.

tancnle commented:
Thank you for bearing with me @allanfarrell 🙏🏼 The relative path to the wrapper script seems to be misinterpreted in Windows. I have tweaked the script to use absolute path and converted it to shell script instead of bash to see if it makes any difference.

allanfarrell commented:
![image](https://user-images.githubusercontent.com/22616875/166137186-4cae9302-43b8-4c50-a7bb-af9e3a70163e.png)

tancnle committed:
fix: convert to bash for wrapper script

shell script does not support setting option flag (i.e. set) and causes
invalid syntax error.

tancnle commented:
Ah, different error this time. We are making some progress, at least the `translate_wrapper` file is now found and executed. I have fixed up the script syntax to be bash. Another go?

dylanmedlin commented:
I got the same error

![image](https://user-images.githubusercontent.com/1919022/166139390-a311e0e8-b39f-4636-9544-d0b0b58e1cff.png)

dylanmedlin commented:
You need to change the translate_wrapper to LF line endings for it to work

![image](https://user-images.githubusercontent.com/1919022/166139884-8b96a0de-980c-4fee-9349-f54c750da408.png)

dylanmedlin commented:
Maybe we should change the .gitattributes file to include all files, not just Ruby files

dylanmedlin committed:
Update .gitattributes

Set all files to have LF line endings

dylanmedlin commented:
@allanfarrell see if it works for you now

tancnle commented:
> Maybe we should change the .gitattributes file to include all files, not just Ruby files

Git client on Windows must have been auto-converting line ending on check out 🤔 Out of curiosity, what is the output this command on your console

```sh
git config core.autocrlf
```

I think https://github.com/thoth-tech/splashkit-translator/pull/5/commits/a6a181f94b34f429e05583ba213271d896f66a0b will ensure local config such as `autocrlf` is overridden and consistently set as `LF`.

dylanmedlin commented:
The output is 'true' for me, which is why it converts to CRLF automatically

dylanmedlin commented:
Correct https://github.com/thoth-tech/splashkit-translator/commit/a6a181f94b34f429e05583ba213271d896f66a0b will override the local settings

allanfarrell commented:
Tested and working!
![image](https://user-images.githubusercontent.com/22616875/166140595-8bbbe63f-d882-4e5f-bff2-aed0bcf2f2cf.png)
